htdocs/don/class/donstats.class.php

```diff
@@ -76,17 +81,31 @@ public function __construct($db, $socid, $mode, $userid = 0)
 		$this->socid = ($socid > 0 ? $socid : 0);
 		$this->userid = $userid;
 		$this->cachefilesuffix = $mode;
+		$this->join = '';
+
+		if ($status === '0' || $status === '1' || $status === '2') {
```

Comment:
Doing a strict string comparisson on $status that is a string will not yield the expected result when it is indeed an int.